tracing-journald/tests/journal.rs

```diff
@@ -0,0 +1,81 @@
+use std::collections::HashMap;
+use std::process::Command;
+
+use tracing::{error, info, warn};
+use tracing_journald::Subscriber;
+use tracing_subscriber::subscribe::CollectExt;
+use tracing_subscriber::Registry;
+
+fn with_journald(f: impl FnOnce()) {
+    let sub = Registry::default().with(Subscriber::new().unwrap().with_field_prefix(None));
+    tracing::collect::with_default(sub, f);
+}
+
+/// Read from journal with `journalctl`.
+///
+/// `test_name` is the randomized name of the test being run, and gets
+/// added as `TEST_NAME` match to the `journalctl` call, to make sure to
+/// only select journal entries originating from and relevant to the
+/// current test.
+fn read_from_journal(test_name: &str) -> Vec<HashMap<String, String>> {
+    let stdout = String::from_utf8(
+        Command::new("journalctl")
+            .args(&["--user", "--output=json"])
+            // Filter by the PID of the current test process
+            .arg(format!("_PID={}", std::process::id()))
+            // tracing-journald logs strings in their debug representation
+            .arg(format!("TEST_NAME={:?}", test_name))
+            .output()
+            .unwrap()
+            .stdout,
+    )
+    .unwrap();
+
+    stdout
+        .lines()
+        .map(|l| serde_json::from_str(l).unwrap())
+        .collect()
+}
+
+#[test]
+fn simple_message() {
+    with_journald(|| info!(test.name = "simple_message", "Hello World"));
+
+    let messages = read_from_journal("simple_message");
+    assert_eq!(messages.len(), 1);
+
+    let message = &messages[0];
+    assert_eq!(message["MESSAGE"], "Hello World");
+    // tracing-journald logs string fields in their debug representation, hence the extra quotes
+    assert_eq!(message["TEST_NAME"], "\"simple_message\"");
```

Comment:
@hawkw @Ralith While porting this test over from libsystemd.rs I noticed that the journald subscriber forwards strings in additional fields in their `Debug` representation, hence these extra quotes here.

Is this intended?  Curiously `MESSAGE` isn't in `Debug`, so it strikes me as not entirely consistent :thinking:  It also makes it hard for downstream journal consumers to extract the original string out of an extra field (e.g. through journalctl's JSON output) without "unparsing" Rust's `Debug` representation of strings.

I think personally I'd prefer send strings in extra fields as is, i.e. as strings, not as `Debug` strings, over to the journal.  Should I open a separate bug ticket for this, or can I directly make another merge request?

Reply:
The current behavior is intended, though I'm open to considering alternative approaches in a new issue. `MESSAGE` is always a string, so it's encoded directly, whereas structured values may have any type; if we special-cased strings, then it would e.g. be impossible to distinguish the string `"1"` from the integer `1`.

Reply:
@Ralith I'll make a new issue for this :slightly_smiling_face:

Reply:
@Ralith I made https://github.com/tokio-rs/tracing/issues/1710